includes/admin_import.php

```diff
@@ -167,10 +167,40 @@ function tsml_import_page()
 			]);
 			if (is_array($response) && !empty($response['body']) && ($body = json_decode($response['body'], true))) {
 
-				//if already set, hard refresh
+				/*When a data source already exists we want to apply changes detected to the local db */  
+
+				//initialize variables 
+				$feed_updates = $db_ids_to_delete = $message_lines = [];
+				$message = "<br><h2>Local Database Updates Being Applied From the $data_source_name Feed</h2><br>";
+				$data_source_last_import = null;
+
+				//if already set, we want to remove only db records which have been changed or removed from the feed
 				if (array_key_exists($data_source_url, $tsml_data_sources)) {
-					tsml_delete(tsml_get_data_source_ids($data_source_url));
+
+					$tsml_data_sources = get_option('tsml_data_sources', array());
+					$data_source_last_import = (int) $tsml_data_sources[$data_source_url]['last_import'];
+
+					//get updated feed records only
+					$feed_updates = tsml_get_import_changes_only($body, $data_source_url, $data_source_last_import, $db_ids_to_delete, $message_lines);
+
+					 /* Drop data source database records which have been updated or removed from the feed */
+					foreach ($db_ids_to_delete as $id) {
+						tsml_delete_by_id($id);
+					} 
 					tsml_delete_orphans();
+					//rebuild cache
+					tsml_cache_rebuild();
+
+					if (count($feed_updates) === 0) {
+						$last_update_date = date('M j, Y  g:i a', $data_source_last_import);
+						exit('<br><h2>No Updates Available For Your ' . $data_source_name . ' Feed</h2>Your local database meeting list records derived from the ' . $data_source_name . ' feed are already in sync!<br>Press the browser back button to return to the previous screen.<br>' );
+					}
+					else {
+						$message .= "<table border='1' style='width:600px;'><tbody><tr><th>Update Mode</th><th>Meeting Name</th><th>Day of Week</th><th>Last Updated</th></tr>";
+						$message .= implode('', $message_lines);
+						$message .= "</tbody></table>";
+						print_r($message);
```

Comment:
is this `print_r` for debugging?

Reply:
No. Using to display feedback.

Reply:
Dropped the print_r and now simply use tsml_alert($message) for the html table display.

Reply:
great! this makes the user feedback consistent with how we report other results like JSON parse errors, etc